File: LostAnimalsAPI/LostAnimalsAPI/Contracts/Requests/UpdatePostRequest.cs
using LostAnimalsAPI.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LostAnimalsAPI.Contracts.Requests
{
    public class UpdatePostRequest
    {
        public long PostId { get; set; }
        public SpeciesLookup Species { get; set; }
        public BreedLookup Breed { get; set; }
        public ColorLookup Color { get; set; }
        public Size Size { get; set; }
        public PostType PostType { get; set; }
        public string Content { get; set; }
        public Location Location { get; set; }
        public DateTime LostTime { get; set; }
        public byte[] ImageSource { get; set; }
    }
}

File: LostAnimalsAndroid/app/src/main/java/com/pbl/animals/models/contracts/requests/UpdatePostRequest.java
package com.pbl.animals.models.contracts.requests;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.pbl.animals.models.Location;
import com.pbl.animals.models.inner.BreedLookup;
import com.pbl.animals.models.inner.ColorLookup;
import com.pbl.animals.models.inner.PostType;
import com.pbl.animals.models.inner.Size;
import com.pbl.animals.models.inner.SpeciesLookup;

import java.util.Date;

public class UpdatePostRequest {
    public long postId;
    public SpeciesLookup species;
    public BreedLookup breed;
    public ColorLookup color;
    public Size size;
    public PostType postType;
    public String content;
    public Location location;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    public Date lostTime;

    public byte[] imageSource;
}
